crates/sui-graphql-client/src/lib.rs

```diff
@@ -1205,6 +1211,74 @@ impl Client {
             Ok(None)
         }
     }
+
+    // ===========================================================================
+    // SuiNS
+    // ===========================================================================
+    /// Get the address for the provided Suins domain name.
+    pub async fn resolve_suins_to_address(&self, domain: &str) -> Result<Option<Address>, Error> {
+        let operation = ResolveSuiNSQuery::build(ResolveSuiNSQueryArgs { name: domain });
+        let response = self.run_query(&operation).await?;
+
+        if let Some(errors) = response.errors {
+            return Err(Error::msg(format!("{:?}", errors)));
+        }
+
+        Ok(response
+            .data
+            .and_then(|d| d.resolve_suins_address)
+            .map(|a| a.address))
+    }
+
+    /// Get the SuiNS registrations for the provided address, returning a page of (Object, Domain)
+    /// tuples.
+    pub async fn suins_registrations(
+        &self,
+        address: Address,
+        pagination_filter: PaginationFilter<'_>,
+    ) -> Result<Page<(Object, Domain)>, Error> {
```

Comment:
Do we think this is even a useful query?

Reply:
The way I understand it is that it could be useful to ask for a list of domains registered to an address. 
We can do this
```
{
  resolveSuinsAddress(domain: "manos.sui") {
    address
  }
}
``` 
So we should also be able to do this
```
{
  address(
    address: "0xe0b97bff42fcef320b5f148db69033b9f689555348b2e90f1da72b0644fa37d0"
  ) {
    suinsRegistrations {
      nodes {
        address
        domain
      }
    }
  }
}
```
Where we'd get something like this:
```
{
  "data": {
    "address": {
      "suinsRegistrations": {
        "nodes": [
          {
            "address": "0x50343e4f16687debdb044a94c9798fa77af04d48aaa11800ebfaabdef382514d",
            "domain": "manolis.sui"
          },
          {
            "address": "0x8ff34f4e38ff98d49d1a0d947e2bc6cb158367044be4a4474513a4fc01fb3356",
            "domain": "152.sui"
          },
          {
            "address": "0xf1257c5d0904a39980feee979b0f66cc55659523f2a0d9db44b27dc04297c7de",
            "domain": "manos.sui"
          }
        ]
      }
    }
  }
}
```
I wouldn't see any other way to get such information. Now if this is useful or not, I will ask Manos. I am not familiar with the suins use-cases of reading such information from chain.

Reply:
@bmwill you were right, this is not needed. But what we need is this, so will add it now. 
```
{
  address(address:"") {
      defaultSuinsName
  }
}
```